nuxeo-runtime/nuxeo-launcher-commons/src/main/java/org/nuxeo/launcher/config/ConfigurationGenerator.java

```diff
@@ -289,8 +291,16 @@
      */
     public static final String JAVA_OPTS_PROP = "launcher.java.opts";
 
+    public static final String VERSIONED_REGEX = "(-[0-9]+.[0-9]+.[0-9]+)?";
```

Comment:
`.` should be `\\.` really if you want to match a dot exactly. I'd use a regexp `"(-\\d+(\\.\\d+)*)?"` to simplify.